conmon/conmon.c

```diff
@@ -780,6 +819,28 @@ int main(int argc, char *argv[])
 						pexit("Failed to add client socket fd to epoll");
 					}
 					ninfo("Accepted connection");
+				} else if (!exec && evlist[i].data.fd == ctlfd) {
+					num_read = read(ctlfd, buf, BUF_SIZE);
```

Comment:
This doesn't seem quite right. You're just assuming that the read will only get one, and and entire line. You may very well get partial lines or multiple lines in one read op.